### PR TITLE
Multilib error fixes

### PR DESCRIPTION
**Pull request by statham-arm**
This PR contains two separate patches updating the "custom error message" feature in `multilib.yaml` (#105684):

* Change the YAML keyword `FatalError` to `Error`, as @petrhosek requested after the previous PR was already landed
* Improve ergonomics of the error report: it's better to have it after the long list of available multilibs, not before, so that it appears in the most sensible place for a fatal error message and doesn't get scrolled off the user's screen.

### CODE REVIEW

File: clang/lib/Driver/Multilib.cpp [hunk 1]
```diff
@@ -124,12 +124,11 @@ bool MultilibSet::select(const Driver &D, const Multilib::flags_list &Flags,
     }
 
     // If this multilib is actually a placeholder containing a fatal
-    // error message written by the multilib.yaml author, display that
-    // error message, and return failure.
-    if (M.isFatalError()) {
-      D.Diag(clang::diag::err_drv_multilib_custom_error) << M.getFatalError();
-      return false;
-    }
+    // error message written by the multilib.yaml author, then set a
```
Comment (smithp35) [inline clang/lib/Driver/Multilib.cpp:127]: On the line above, should that be "containing an error message written ... " rather than fatal error?

**Review by smithp35 — COMMENTED** [review]
Change looks good to me. A few small comment nits.

File: clang/lib/Driver/Multilib.cpp [hunk 2]
```diff
@@ -217,15 +215,15 @@ struct MultilibSetSerialization {
...
     io.mapRequired("Flags", V.Flags);
     io.mapOptional("Group", V.Group);
   }
   static std::string validate(IO &io, MultilibSerialization &V) {
-    if (V.Dir.empty() && V.FatalError.empty())
-      return "one of the 'Dir' and 'FatalError' keys must be specified";
-    if (!V.Dir.empty() && !V.FatalError.empty())
-      return "the 'Dir' and 'FatalError' keys may not both be specified";
+    if (V.Dir.empty() && V.Error.empty())
+      return "one of the 'Dir' and 'atalError' keys must be specified";
```
Comment (smithp35) [inline clang/lib/Driver/Multilib.cpp:224]: The 'atalError' looks like a typo. It probably means we're missing a test for that message too.

Comment (statham-arm) [reply clang/lib/Driver/Multilib.cpp]: Ha – that's what I get for checking I'd caught them all by grepping for `Fatal`. The grep missed this one! (And also the lower-case one you pointed out before.)

**Review by smithp35 — APPROVED** [review]
Thanks for the updates. I don't have any more comments. I've approved the patch on my side. Will be worth waiting for a bit to see if any other reviewers have any feedback.

**Review by petrhosek — APPROVED** [review]
Thanks for making the change!

**Pull request closed without merging**